# Generated by Django 4.2.20 on 2025-04-19 12:45

import django.contrib.gis.db.models.fields
import django.core.validators
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('forecast', '0001_initial'),
    ]

    operations = [
        migrations.CreateModel(
            name='Zone',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=255, null=True)),
                ('geom', django.contrib.gis.db.models.fields.GeometryField(blank=True, geography=True, null=True, srid=4326)),
                ('active', models.BooleanField(default=True)),
                ('category', models.CharField(blank=True, choices=[('polygon', 'Plolygone'), ('commune', 'Commune'), ('province', 'Province'), ('region', 'RÃ©gion'), ('ville', 'Ville'), ('axe', 'Axe'), ('point', 'Point'), ('divers', 'Divers')], max_length=10, null=True)),
                ('rayon', models.FloatField(blank=True, default=0, null=True, validators=[django.core.validators.MinValueValidator(0.0)])),
            ],
        ),
        migrations.AddConstraint(
            model_name='zone',
            constraint=models.UniqueConstraint(fields=('name', 'category'), name='unique_name_category'),
        ),
    ]
